### ISSUE TITLE
Cannot build application

### ISSUE THREAD

**Issue by ikarelin** [parse-community/Parse-SDK-Flutter#1004]
### New Issue Checklist

- [x ] I am not disclosing a [vulnerability](https://github.com/parse-community/Parse-SDK-Flutter/security/policy).
- [ x] I am not just asking a [question](https://github.com/parse-community/.github/blob/main/SUPPORT.md).
- [ x] I have searched through [existing issues](https://github.com/parse-community/Parse-SDK-Flutter/issues?q=is%3Aissue).
- [ x] I can reproduce the issue with the latest version of [Parse Server](https://github.com/parse-community/parse-server/releases) and the [Parse Flutter SDK](https://github.com/parse-community/Parse-SDK-Flutter/releases). 

### Issue Description
Cannot build flutter application with parse server plugin 

### Steps to reproduce
Run application build in IntelliJ IDEA

### Actual Outcome
 Running Xcode build...                                                  
Xcode build done.                                           34.8s
Failed to build iOS app
Error (Xcode): ../../.pub-cache/hosted/pub.dev/parse_server_sdk_flutter-8.0.0/lib/parse_server_sdk_flutter.dart:126:53: Error: The argument type
'ParseConnectivityResult Function(ConnectivityResult)' can't be assigned to the parameter type 'ParseConnectivityResult Function(List<ConnectivityResult>)'.

Encountered error while building for device.

### Expected Outcome
Normal build and run application

### Environment

Parse Flutter SDK
- SDK version: `parse_server_sdk_flutter-8.0.0`
- Flutter version: `Flutter 3.22.3 • channel stable`
- Dart version: `Dart SDK version: 3.4.4 (stable)`
- Operating system version: `Mac OS Sonoma 14.5 (23F79)`

Server
- Parse Server version: `6.5.4`

### Logs
Running Xcode build...
Xcode build done.                                           31.4s
Failed to build iOS app
Error (Xcode): ../../.pub-cache/hosted/pub.dev/parse_server_sdk_flutter-8.0.0/lib/parse_server_sdk_flutter.dart:126:53: Error: The argument type 'ParseConnectivityResult Function(ConnectivityResult)' can't be assigned to the parameter type 'ParseConnectivityResult Function(List<ConnectivityResult>)'.

Could not build the application for the simulator.
Error launching application on iPhone 15 Pro Max.

**Comment by mtrezza** [comment]
Do you have any code to share that is causing this?

**Comment by ikarelin** [comment]
It's not in the code, the code cannot compile. It's in the plugin code:

[  +96 ms] Failed to build iOS app
[   +2 ms] Error (Xcode): [+1408 ms] ../../.pub-cache/hosted/pub.dev/**parse_server_sdk_flutter-8.0.0/lib/parse_server_sdk_flutter.dart:126:53**: Error: The argument
type
           'ParseConnectivityResult Function(ConnectivityResult)' can't be assigned to the parameter type 'ParseConnectivityResult
Function(List<ConnectivityResult>)'.

**Comment by mtrezza** [comment]
So you are only adding the Parse SDK to a new project without any code and then this error occurs during compile time?

**Comment by ikarelin** [comment]
No, empty Flutter project works fine with just parse_server_sdk_flutter installed.
I've found the problem is with connectivity_plus flutter plugin, when I'm disabling it in my project, build going normally.

**Comment by ikarelin** [comment]
Seems the also project build normally with connectivity_plus version 5.0.2, but not with the latest 6.0.4 version

**Comment by chadpav** [comment]
@ikarelin You might look at a PR/Issue I had last summer and see if it is the root cause of your problem. There was a breaking change in `connectivity_plus` in 6.x and the Flutter SDK isn't compatible.

https://github.com/parse-community/Parse-SDK-Flutter/pull/1002

**Comment by kirkmorrow** [comment]
@ikarelin is this still a problem?
